tests/test_build.py

```diff
@@ -40,6 +40,21 @@ def test_respects_none_overrides() -> None:
     assert result.hobbies is None
 
 
+def test_allows_nested_optional_field_overrides() -> None:
+    class MyChildModel(BaseModel):
+        name: str
+
+    class MyParentModel(BaseModel):
+        child: MyChildModel | None
+
+    class MyParentModelFactory(ModelFactory):
+        __model__ = MyParentModel
+
+    MyParentModelFactory.seed_random(2)
+    result = MyParentModelFactory.build(child={"name": "test"})
+    assert result.child is not None
+
+
```

Comment:
I think this might be better to move this into the `test_passing_build_args_to_child_factories.py`.

Reply:
https://github.com/litestar-org/polyfactory/pull/420/commits/dec60a4f8a233f18c9951305157cb6f237a26b54